### PR TITLE
#171815577 improve code maintainability

### PR DESCRIPTION
**Pull request by Kagorora**
#### What does this PR do?

- This PR will improve code maintainability by removing duplication and code smells

#### Description of Task to be completed?

- will remove duplication on these specific files:

    src/controllers/searchRequest.js
    src/controllers/notifications.js

- will reduce the number of files on 

    function `login` in file  src/controllers/users
    function `registerUser` src/controllers/users.js
    function  `rejectRequest ` src/controllers/request.js
   function `checkAuthorizedCommenter` src/middlewares/commenter.js
    function `signUp` src/middlewares/newUser.js
    function `getRequestStatus` src/controllers/notifications.js

#### How should this be manually tested?

- consult the badge using this link [link](https://codeclimate.com/github/andela/knights-bn-backend)

#### Any background context you want to provide?

- N/A

#### What are the relevant pivotal tracker stories?

- [171815577](https://www.pivotaltracker.com/story/show/171815577)

#### Screenshots (if appropriate)

- N/A

#### Questions:

- N/A

### CODE REVIEW

File: src/helpers/sendEmail.js [hunk 1]
```diff
@@ -21,3 +21,24 @@ const sendEmail = async (businessEmail, requester, request) => {
   sendGrid.send(msg);
 };
 export default sendEmail;
+
+// sgMail.setApiKey(process.env.BN_API_KEY);
```
Comment (T2Wil) [inline src/helpers/sendEmail.js]: hey @Kagorora , I think these comments should be deleted!?

File: src/middlewares/commenter.js [hunk 2]
```diff
@@ -4,34 +4,17 @@ import UserQuery from '../helpers/userQueries';
...
-      error: 'request does not exist',
-    });
+    return res.status(422).json({ status: 422, error: 'please provide the Id of the request, Id must be a number' }); 
   }
+  const requestInfo = await models.Request.findOne({ where: { id: requestId } });
+  if (!requestInfo) return res.status(404).json({ status: 404, error: 'request does not exist' });
   const { requesterId } = requestInfo;
   const requester = await UserQuery.getUser('id', requesterId);
   const { lineManager } = requester.dataValues;
   const Manager = await UserQuery.getUser('email', lineManager);
```
Comment (T2Wil) [inline src/middlewares/commenter.js]: @Kagorora , is there a reason you capitalized this` manager` object?

File: src/middlewares/validateDate.js [hunk 3]
```diff
@@ -65,15 +64,16 @@ export const checkIfRequestExists = async (req, res, next) => {
...
+  const pendingRequestLength = pendingRequest.length;
+  if (pendingRequestLength !== 0) {
     return res.status(409).json({ satatus: 409, error: 'You still have pending request just wait for approval' });
   }
 };
 
 export const validateRequestDate = (req, res, next) => {
   const insertedDate = new Date(req.body.returnDate);
-  if (insertedDate.getFullYear() < new Date().getFullYear() || insertedDate.getFullYear() > new Date().getFullYear() + 2) {
+  if (insertedDate.getFullYear() < new Date().getFullYear() || insertedDate.getFullYear()
```
Comment (Niyonsengaeric) [inline src/middlewares/validateDate.js]: can you remove also this duplicate!

File: src/tests/accommodation/createAccommodationFacility.test.js [hunk 4]
```diff
@@ -238,64 +202,78 @@ export const editAccommodations = () => {
...
   });
 };
 
-export const violatingDatabase = () => {
-  describe('accommodation ', () => {
-    before((done) => {
-      localStorage.setItem('token', travelToken);
-      done();
-    });
+// ---------------------------------------------- SUPPLIER
```
Comment (T2Wil) [inline src/tests/accommodation/createAccommodationFacility.test.js]: @Kagorora , remove this comment too.

File: src/helpers/sendEmail.js (hunk 1 above)
Comment (Kagorora) [reply src/helpers/sendEmail.js]: @T2Wil I have removed it

File: src/middlewares/commenter.js (hunk 2 above)
Comment (Kagorora) [reply src/middlewares/commenter.js]: fixed as well, you are right only class name should be capitalized

File: src/middlewares/validateDate.js (hunk 3 above)
Comment (Kagorora) [reply src/middlewares/validateDate.js]: @Niyonsengaeric That is not duplication as it seems

File: src/tests/accommodation/createAccommodationFacility.test.js (hunk 4 above)
Comment (Kagorora) [reply src/tests/accommodation/createAccommodationFacility.test.js]: fixed

**Review by T2Wil — APPROVED** [review]
Nice work @Kagorora !

**Review by Niyonsengaeric — APPROVED** [review]

File: src/controllers/users.js [hunk 5]
```diff
@@ -10,24 +10,22 @@ import generatePswd from '../helpers/randomPswd';
...
+        firstName, lastName, gender, passportNumber, email, password,
       } = req.body;
       const token = generateToken({
         firstName, lastName, gender, passportNumber, email, password,
       });
       let host;
-      if (process.env.NODE_ENV === 'development') {
-        host = process.env.LOCAL_HOST;
-      } else {
+      if (process.env.NODE_ENV === 'development') { host = process.env.LOCAL_HOST; } else {
```
Comment (kagabof) [inline src/controllers/users.js]: Please check that code climate issue.

File: src/middlewares/validateDate.js (hunk 3 above)
Comment (kagabof) [reply src/middlewares/validateDate.js]: Please check that code climate issue.

File: src/middlewares/validateDate.js [hunk 6]
```diff
@@ -140,11 +139,13 @@ export const validateCityDate = (req, res, next) => {
   if (typeof req.body.cities !== 'undefined') {
     req.body.cities.forEach((city) => {
       const insertedDate = new Date(city.to);
-      if (insertedDate.getFullYear() < new Date().getFullYear() || insertedDate.getFullYear() > new Date().getFullYear() + 2) {
+      if (insertedDate.getFullYear() < new Date().getFullYear() || insertedDate.getFullYear()
```
Comment (kagabof) [inline src/middlewares/validateDate.js]: Please check that code climate issue.

**Review by kagabof — CHANGES_REQUESTED** [review]

**Review by kagabof — APPROVED** [review]
Great job.

**Pull request merged**